src/template_variables/mod.rs

```diff
@@ -3,7 +3,68 @@ mod crate_type;
 mod os_arch;
 mod project_name;
 
+use crate::{config::ConfigValues, emoji, Args};
+
+use anyhow::Result;
+use console::style;
+use regex::Regex;
+use std::{collections::HashMap, fs, path::Path};
+use toml::Value;
+
 pub(crate) use authors::{get_authors, Authors};
 pub(crate) use crate_type::CrateType;
 pub(crate) use os_arch::get_os_arch;
 pub(crate) use project_name::ProjectName;
+
+pub(crate) fn resolve_template_values(args: &Args) -> Result<HashMap<String, Value>> {
+    let mut template_values = args
+        .template_values_file
+        .as_ref()
+        .map(|p| Path::new(p))
+        .map_or(Ok(Default::default()), |path| {
+            read_template_values_file(path)
+        })?;
+
+    add_cli_defined_values(&mut template_values, args)?;
+
+    Ok(template_values)
+}
+
+fn read_template_values_file(path: &Path) -> Result<HashMap<String, Value>> {
+    match fs::read_to_string(path) {
+        Ok(ref contents) => toml::from_str::<ConfigValues>(contents)
+            .map(|v| v.values)
+            .map_err(|e| e.into()),
+        Err(e) => anyhow::bail!(
+            "{} {} {}",
+            emoji::ERROR,
+            style("Values File Error:").bold().red(),
+            style(e).bold().red(),
+        ),
+    }
+}
+
+fn add_cli_defined_values(template_values: &mut HashMap<String, Value>, args: &Args) -> Result<()> {
+    let regex = Regex::new(r"(\w+)=(.+)").unwrap();
```

Comment:
should be better allow a more advanced identifier on the left hand side? 

e.g. something like `project-description` is not covered at the moment.

Reply:
it is now - as well as a few unittests ✔️